### ISSUE TITLE
OperatorSetProto.onnx: the instance of immutable operator set domain ai.onnx and ai.onnx.ml

### ISSUE THREAD

**Issue by elfman2** [onnx/onnx#6163]
### System information

any version

### What is the problem that this feature solves?

The https://github.com/onnx/onnx/blob/main/onnx/onnx-operators.proto defines the opset meta model. It does not seem to be instanciated (OperatorSetProto.onnx ?)

It seems that the runtime redefines the opsets in the header file :https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorVersions.h instead of loading an onnx OperatorSetProto file.

Would ease the third party tools development to get a standard proto file for OperatorSet.

### Alternatives considered

I looked in several tools repos and did not find any protobuf instance of OperatorSetProto

### Describe the feature

Avoid duplication of immutable operator versions per opset in every runtime, and tool.

### Will this influence the current api (Y/N)?

N

### Feature Area

Any tool which needs to identify the operator version in a model.

### Are you willing to contribute it (Y/N)

Yes

**Comment by gramalingam** [comment]
Correct. In the beginning, there was a desire to have operator specs in a portable, serialized, format. However, this was never implemented.

There is at least one aspect that is hard to serialize: an operator's shape-inference logic, which, in the general case, can be arbitrary C++ code.
